test/e2e/framework/util.go

```
@@ -2421,7 +2421,29 @@ func WaitForAllNodesSchedulable(c clientset.Interface, timeout time.Duration) er
 	})
 }
 
-func GetTTLAnnotationFromNode(node *v1.Node) (time.Duration, bool) {
+func GetPodSecretUpdateTimeout(c clientset.Interface) time.Duration {
+	// With SecretManager(ConfigMapManager), we may have to wait up to full sync period +
+	// TTL of secret(configmap) to elapse before the Kubelet projects the update into the
+	// volume and the container picks it up.
+	// So this timeout is based on default Kubelet sync period (1 minute) + maximum TTL for
+	// secret(configmap) that's based on cluster size + additional time as a fudge factor.
+	secretTTL, found := GetNodeTTLAnnotationValue(c)
+	if !found {
+		Logf("Couldn't get node ttl annotation")
+	}
+	podLogTimeout := 240*time.Second + secretTTL
+	return podLogTimeout
+}
+
+func GetNodeTTLAnnotationValue(c clientset.Interface) (time.Duration, bool) {
+	nodes, err := c.Core().Nodes().List(metav1.ListOptions{})
```

Please don't silently drop errors. Instead:
```
ExpectNoError(err)
if len(nodes.Items) == 0 {
  ExpectNoError(...)
}
```